datalad/distribution/create_publication_target_sshwebserver.py

```diff
@@ -284,13 +285,28 @@ def __call__(sshurl, target=None, target_dir=None,
                             "You will not be able to push to checked out "
                             "branch." % path)
 
-            # enable post-update hook:
+            # create post_update script
+            lgr.info("Enabling git post-update hook ...")
             try:
-                ssh(["mv",
-                     opj(path, ".git/hooks/post-update.sample"),
-                     opj(path, ".git/hooks/post-update")])
+                json_command = '"datalad ls -r --json file ' + path + '"'
+                virtualenv = 'source /home/debanjum/datalad/.env/bin/activate'  # NOT_SCALABLE! custom virtualenv path
+                for post_update_hook in ['"#!/bin/bash"', '"git update-server-info"', virtualenv, json_command]:
+                    ssh(["echo", post_update_hook, '>>', opj(path, '.git/hooks/post-update')])
+                ssh(["chmod", '+x', opj(path, '.git/hooks/post-update')])  # make the script executable
             except CommandError as e:
-                lgr.error("Failed to enable post update hook.\n"
+                lgr.error("Failed to add json creation command to post update hook.\n"
+                          "Error: %s" % e.message)
+
+            # get html for dataset rendering from local datalad repo:
+            lgr.info("Uploading web interface ...")
+            try:
+                import datalad
+                html = opj(dirname(datalad.__file__), "resources/website/index.html")
+                scp_target = sshri.hostname + ":" + sshri.path
+                Runner().run(['scp', '-P', sshri.port, html, scp_target] if sshri.port else ['scp', html, scp_target])
```

Comment:
ideally we should avoid direct scp calls like this.  There is this `ssh` manager which should be our central point of interaction with ssh.  Now you would not take advantage of controlpath and not sure how `sshri.port` would play out here if not specified in the url.  So ideally I think we should provide  `SSHConnection.copy(path, remote_dest, ...)` to provide 'scp' operation, may be with support for options such as recursion, preserving permissions etc... 
Then it should also be used above instead of the chain of "echo" calls to create post-update hook.  You can generate entire temp file locally and then .copy it there.